### PR TITLE
Add check to ensure milestone

### PR DESCRIPTION
**Pull request by jefferai**
Stolen from https://github.com/hashicorp/vault/pull/18406

### CODE REVIEW

File: .github/workflows/milestone-checker.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# This workflow checks that there is either a 'pr/no-milestone' label applied
+to a PR # or there is a milestone associated with a PR
```
Comment (johanbrandhorst) [inline .github/workflows/milestone-checker.yml:2]: This looks like it's going to cause a yaml parse error

Comment (jefferai) [reply .github/workflows/milestone-checker.yml]: Huh, didn't realize it auto wrapped on paste, thanks!

**Comment by jefferai** [comment]
Note: Since the script uses the config json from main it won't pass until it's merged.

File: .github/workflows/milestone-checker.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    types: [opened, synchronize, labeled, unlabeled]
+    # Runs on PRs to main and release branches
+    branches:
+      - main
+      - release/**
+
+jobs:
+  # checks that a milestone entry is present for a PR
+  milestone-check:
+    # If there  a `pr/no-milestone` label we ignore this check
```
Comment (johanbrandhorst) [inline .github/workflows/milestone-checker.yml:17]: Nit
```suggestion
    # If there is a `pr/no-milestone` label we ignore this check
```

File: .github/workflows/milestone-checker.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    if: "!contains(github.event.pull_request.labels.*.name, 'pr/no-milestone')"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Actions
+        uses: actions/checkout@v2
+        with:
+          repository: "grafana/grafana-github-actions"
+          path: ./actions
+          ref: main
```
Comment (johanbrandhorst) [inline .github/workflows/milestone-checker.yml:26]: Can we lock this to a ref to prevent supply chains attack in the grafana github actions?

**Review by johanbrandhorst — APPROVED** [review]
🙏🏻 lets hope it builds on `main` 😄

**Pull request merged**